### PR TITLE
feat: APP-183 attach signature to data post + APP-240

### PR DESCRIPTION
**Pull request by blushi**
## Description

https://regennetwork.atlassian.net/browse/APP-183?atlOrigin=eyJpIjoiYTczMWFmNjQ1NjhkNGU2Y2FiYTk2NzYzM2E5YWJkMjIiLCJwIjoiaiJ9

https://regennetwork.atlassian.net/browse/APP-240?atlOrigin=eyJpIjoiZmNkYmEwMTIwNjM0NGM5MjlhZjhmNGUxN2JjZmYwMzMiLCJwIjoiaiJ9

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

From https://deploy-preview-2423--regen-marketplace.netlify.app/
1. Log in with a web2 account and create a post, you should see anchoring hash in the success modal
2. Log in with a web3 account and create a post. After entering post data, you should see a modal to sign the post or skip this step. Test both options. If signing a post successfully, you'll see the attesting hash in the success modal too.

The button at the bottom of the success modal now goes to the newly created post page.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by blushi** [comment]
@erikalogie @clevinson see testing instructions

**Comment by blushi** [comment]
@erikalogie could you test this again?

**Comment by erikalogie** [comment]
It is working well for me now!

**Comment by blushi** [comment]
> It is working well for me now! Just double-check the font size for the hashes in the confirmation popup after a user has signed, does not match the comps or the other confirmation popups.

@erikalogie could you have another look?

**Comment by erikalogie** [comment]
Looks great, one last tiny detail, these don't seem to be perfectly left-aligned on mobile:
Regen Marketplace : Buy, trade and retire carbon credits | Regen Marketplace : Buy, trade and retire carbon credits 2024-08-08 08-56-34

**Comment by clevinson** [comment]
This looks great!

Screenshot 2024-08-08 at 3 09 00 PM
Screenshot 2024-08-08 at 8 57 42 PM

I just have a few copy suggestions.

When no signature is provided, there's no title in the modal that pops up. Additionally I think the "Attest" title should be a bit more clear.

Can we instead have the title always say "Create Data Post", similar to how when creating a project it says "Create Project"

Then, in the section below I think we should change the following:
1. "Hash" section header should be "Blockchain Record"
2. "Anchor" should be "Create Post"
3. "Attest" should be "Signature"

I think those will be more clear to non web3 users. @erikalogie @blushi let me know what y'all think? In general I think we can leave "Anchor" and "Attest" out of the marketplace UIs, and have them only appears in API documentation and Mintscan.

**Comment by erikalogie** [comment]
> This looks great!
> 
> Screenshot 2024-08-08 at 3 09 00 PM Screenshot 2024-08-08 at 8 57 42 PM
> I just have a few copy suggestions.
> 
> When no signature is provided, there's no title in the modal that pops up. Additionally I think the "Attest" title should be a bit more clear.
> 
> Can we instead have the title always say "Create Data Post", similar to how when creating a project it says "Create Project"
> 
> Then, in the section below I think we should change the following:
> 
> 1. "Hash" section header should be "Blockchain Record"
> 2. "Anchor" should be "Create Post"
> 3. "Attest" should be "Signature"
> 
> I think those will be more clear to non web3 users. @erikalogie @blushi let me know what y'all think? In general I think we can leave "Anchor" and "Attest" out of the marketplace UIs, and have them only appears in API documentation and Mintscan.

All sounds good to me. If we are going to change "hash" here then we should probably change it on all the other confirmation popups too for consistency.

**Review by r41ph — APPROVED** [review]

**Comment by blushi** [comment]
@erikalogie @clevinson ready for another review, I've applied the copy suggestions.

**Comment by blushi** [comment]
@clevinson @erikalogie we also now have this "NOTE: As posts are anchored to the blockchain, they are not editable once published" at the bottom of the post creation form. Should we rephrase it to get rid of "anchored"?

**Comment by erikalogie** [comment]
> https://deploy-preview-2423--regen-marketplace.netlify.app/

Yes, I think @clevinson will likely have a good suggestion here.

**Comment by erikalogie** [comment]
> @erikalogie @clevinson ready for another review, I've applied the copy suggestions.

The copy looks like nonsense again: 
Screenshot 2024-08-27 at 10 27 10 AM

**Comment by blushi** [comment]
> > @erikalogie @clevinson ready for another review, I've applied the copy suggestions.
> 
> The copy looks like nonsense again: Screenshot 2024-08-27 at 10 27 10 AM

Sorry this is when I forget to generate the .po files for the translation now that we have it set up in the app. I'll fix that.

**Comment by blushi** [comment]
> Sorry this is when I forget to generate the .po files for the translation now that we have it set up in the app. I'll fix that.

@erikalogie this should be fixed now
